addon/mixins/node-actions.js

```diff
@@ -0,0 +1,100 @@
+import Ember from 'ember';
+
+export default Ember.Mixin.create({
+    node: null,
+    model: null,
+    _node: Ember.computed.or('node', 'model'),
+
+    /** Maps all node contributors to format {contribID: Contributor}
+     **/
+    _generateContributorMap(contributors) {
+        var contribMap = {};
+        contributors.forEach(contrib => contribMap[contrib.id] = contrib);
+        return contribMap;
+    },
+    _affiliateNode(node, institution) {
+        node.get('affiliatedInstitutions').pushObject(institution);
+        return node.save();
+    },
+    actions: {
+        updateNode(title, description, category, isPublic) {
+            var node = this.get('_node');
+            if (title) {
+                node.set('title', title);
+            }
+            if (category) {
+                node.set('category', category);
+            }
+            if (description) {
+                node.set('description', description);
+            }
+            if (isPublic !== null) {
+                node.set('public', isPublic);
+            }
+            return node.save();
+        },
+        deleteNode() {
+            var node = this.get('_node');
+            return node.destroyRecord();
```

Comment:
This is a good example of an action that should have a transition after the promise is returned.  Did you remove this because we don't want to enforce in this mixin where to redirect?